QuickDeployment/QuickDeploymentInternetSetup.md

```diff
@@ -0,0 +1,222 @@
+* [Scenario 2: Purchased/Acquired Certificates from CA](#scenario-2-purchasedacquired-certificates-from-ca): This would be an SSL certificate that you purchased or acquired from an external Certificate Authority (CA; e.g. LetsEncrypt).
+* [Scenario 3: Self-Signed SSL Certificates](#scenario-3-self-signed-ssl-certificates): This would be a self-generated SSL certificate, similar to the one that comes with QDE (least-desirable).
+
+Each approach is discussed individually in the sections below.
+
+### Scenario 1: Domain Server Certificates
+
+If you manage a Windows AD Domain, this is going to be the most common scenario. The majority of customers will want to join the QDE server to their domain, and possibly even change the hostname to match their organizational naming conventions. As this process will change your FQDN, this will need to be completed  **_ahead_** of any other changes.
+
+The following prerequisites are required to be in place for later steps to be successful:
+
+1. Ensure that you join the QDE server to the domain **_first_**.
+1. Ensure that the FQDN of the QDE server is resolvable to its IP across your domain.
+
+Once you have your new valid and resolvable FQDN for QDE, you will now want to ensure the domain server certificate associated with that FQDN is utilized for the Nexus and CCM services on QDE.
+
+1. Open the "Certificates - Local Computer" MMC snap-in by pressing the Windows key, and when the Start menu pops up, type certificates. You should now see an option under the "Settings" section that says "Manage computer certificates". Alternatively, open the Run dialog (Windows key + R), type `certlm.msc`, and click "OK".
+1. Under the "Personal" store, you should see a server certificate matching the FQDN of your QDE server. Double-click on the certificate to open it, and under the details tab, copy out the `Thumbprint` value of the certificate. You will require this for later steps.
+1. Now, you can run the `New-SslCertificates.ps1` script and pass the thumbprint you copied above using the `-Thumbprint` parameter. Please run the below command from a PowerShell Administrator window:
+
+    ```powershell
+    Set-ExecutionPolicy Bypass -Scope Process -Force; . C:\choco-setup\files\New-SslCertificates.ps1 -Thumbprint '<YOUR_CERT_THUMBPRINT_HERE>'
+    ```
+
+1. **[OPTIONAL]** If some endpoints are not on the domain, or on a different domain, you will need to ensure that the Nexus and CCM certificates of QDE are copied to the `Local Computer\Trusted People\Certificates` store on those endpoints as well. If the endpoints connecting to Nexus and CCM are also on the same domain, this step may not be necessary as they will inherently trust the domain server certificate of the QDE server.
+
+### Scenario 2: Purchased/Acquired Certificates from CA
```

Comment:
It might be really important to talk about how the FQDN and certificate subject need to match up. What would be my FQDN for a certificate purchased from a CA? If I have a wildcard named "*.chocolatey.org", and I have a machine joined to a domain under "bob.chocolatey.loc", what would be my FQDN? Would it be "bob.chocolatey.org" for over the internet communication versus "bob.chocolately.loc" for in network? 

This probably shows my lack of knowledge in this area, but consider that folks who are setting this up may be in a similar boat.

Reply:
I have included a note and example on this at the beginning:

> Organizations can also opt to purchase or acquire a certificate from an external Certificate Authority (CA; e.g. [LetsEncrypt](https://github.com/win-acme/win-acme)). As mentioned before, you will need to ensure that the "Subject/Common Name" attribute on the SSL certificates matches the FQDN you are using. For example, if you purchase an SSL certificate for your server with the "Subject/Common Name" attribute `chocoserver.example.com`, you will of course need to ensure that you have a DNS record resolving `chocoserver.example.com` to the external IP of your QDE server.